homeassistant/components/androidtv/media_player.py

```diff
@@ -99,6 +100,9 @@
         vol.Optional(CONF_APPS, default=dict()): vol.Schema({cv.string: cv.string}),
         vol.Optional(CONF_TURN_ON_COMMAND): cv.string,
         vol.Optional(CONF_TURN_OFF_COMMAND): cv.string,
+        vol.Optional(CONF_STATE_DETECTION_RULES, default=dict()): vol.Schema(
+            {cv.string: cv.ensure_list}
```

Comment:
Is it possible to make a more extensive config schema for this option to cover all the possible rules?

Reply:
The keys are always app ID's, so `cv.string` covers that.  The values are lists where each item is either

* a string: `'standby'`, `'playing'`, `'paused'`, `'idle'`, `'stopped'`, `'off'`, `'media_session_state'`, or `'audio_state'`
* a map
  * valid keys are: `'standby'`, `'playing'`, `'paused'`, `'idle'`, `'stopped'`, and `'off'`
  * the values are maps
    * valid keys are: `'media_session_state'`, `'audio_state'`, and `'wake_lock_size'`
    * valid values are integers (for `'media_session_state'` and `'wake_lock_size'`) or strings (for `'audio_state'`)

Reply:
I think it's ok to add this option if we build a schema per above.

Reply:
Would it work to do this?

```python
def valid_state_detection_rules(val):
    # raise vol.Invalid if it is invalid and add a message as to why


PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
    {
        ...
        vol.Optional(CONF_STATE_DETECTION_RULES, default=dict()): vol.Schema(
            {cv.string: valid_state_detection_rules}
        ),
    }
)
```

Reply:
Looks good!

Reply:
Done! I also moved the imports to the top of the file because I wanted to use an `androidtv` constant for this config validation.